### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by johangiraud**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* open menu */
+var menuButton = document.querySelector('button.menu_button');
```
Comment (vinyll) [inline script.js]: `const menuButton`

File: script.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+});
+
+/* external link in new tab https://stackoverflow.com/questions/12987963/open-external-links-in-a-new-tab-without-jquery */
+function externalLinks() {
+    for(var c = document.getElementsByTagName("a"), a = 0; a < c.length; a++) {
+        var b = c[a];
+        b.getAttribute("href") && b.hostname !== location.hostname && (b.target = "_blank")
+    }
+}
+externalLinks();
```
Comment (vinyll) [inline script.js]: An alternative proposal would be:
```js
Array.from(document.querySelectorAll('a')).forEach(a => {
  if (a.href.search(/\w+:\/\//) === 0 && a.hostname !== location.hostname) a.setAttribute('target', '_blank')
})
```

However, where do we need that?

File: styleguide.css [hunk 3]
```diff
@@ -246,106 +252,155 @@ input::placeholder {
...
     color: #fff;
 }
+input[type=submit]:hover,
+button:hover {
+    background-color: #fff;
+    color:  #00283C;
+}
 input[type=submit],
-input[type=submit]:hover {
+input[type=submit]:hover,
```
Comment (vinyll) [inline styleguide.css:273]: Why do you need to specify the `:hover` here or on the button?

File: styleguide.css [hunk 4]
```diff
@@ -246,106 +252,155 @@ input::placeholder {
...
+    border: none;
     border-bottom: 0.1rem solid #bde513;
-    padding-left: 0;
     background-color: transparent;
+    padding:1rem 1rem 1rem 0;
 }
 
-[type=checkbox]:not(:checked),
-[type=checkbox]:checked {
+/*[type=checkbox] {
```
Comment (vinyll) [inline styleguide.css]: you may delete if unecessary

File: styleguide.css [hunk 5]
```diff
@@ -364,22 +419,135 @@ p a {
...
-    top:0;
+    top: 0;
     bottom: 0.5rem;
     transform: translateY(0);
     transition: all .3s;
 }
+
+/* Menu */
+
+button.menu_button {
```
Comment (vinyll) [inline styleguide.css]: Why a underscore here and other classes use a dash?

File: styleguide.css [hunk 6]
```diff
@@ -364,22 +419,135 @@ p a {
...
+
+button.menu_button:before,
+button.menu_button:after {
+    content: '';
+    height: 0.3rem;
+    width: 3.5rem;
+    margin: 0;
+    left: 3rem;
+    background: #00283C;
+    position: absolute;
```
Comment (vinyll) [inline styleguide.css:470]: I wonder if we want a position absolute or fixed here.
How does it react if we have scrolled the page? Should we still have the button to access the menu?

File: styleguide.css [hunk 7]
```diff
@@ -364,22 +419,135 @@ p a {
...
+    opacity: 0;
+    z-index: -1;
+    transition: all .3s;
+}
+
+button.menu_button + nav ul {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%,-50%);
```
Comment (vinyll) [inline styleguide.css]: spae after the comma :p

File: script.js (hunk 2 above)
Comment (johangiraud) [reply script.js]: we need it for every external link, mainly inside articles

File: styleguide.css (hunk 3 above)
Comment (johangiraud) [reply styleguide.css:273]: in order to show the user it's a clickable button or input. Doesn't make sense ?

File: styleguide.css (hunk 4 above)
Comment (johangiraud) [reply styleguide.css]: I did

File: styleguide.css (hunk 5 above)
Comment (johangiraud) [reply styleguide.css]: ok, dashes then

File: styleguide.css (hunk 6 above)
Comment (johangiraud) [reply styleguide.css:470]: absolute is enough cause the parent is already fixed

File: script.js (hunk 1 above)
Comment (johangiraud) [reply script.js]: ah ok, what's the diff ?

File: styleguide.css (hunk 4 above)
Comment (vinyll) [reply styleguide.css]: Then you should commit the deleted content, this code appears as comment in the code ;)

File: styleguide.css [hunk 8]
```diff
@@ -364,22 +423,135 @@ p a {
...
+
+button.menu-button:after.active {
+    top: 0;
+}
+
+button.menu-button.active:before {
+    top: 4.6rem;
+    transform: rotate(45deg);
+}
+button.menu-button.active:after {
```
Comment (vinyll) [inline styleguide.css:487]: how is this different from line 488 `button.menu-button:after.active`?

Comment (johangiraud) [reply styleguide.css:487]: `button.menu-button:after.active` line 488 styles the behavior of the menu button, which is not the default button behavior

**Review by vinyll — APPROVED** [review]

**Pull request merged**